rasa/core/agent.py

```diff
@@ -152,25 +159,39 @@ async def _update_model_from_server(
     if not is_url(model_server.url):
         raise aiohttp.InvalidURL(model_server.url)
 
-    model_directory_and_fingerprint = await _pull_model_and_fingerprint(
-        model_server, agent.fingerprint
-    )
-    if model_directory_and_fingerprint:
-        model_directory, new_model_fingerprint = model_directory_and_fingerprint
-        _load_and_set_updated_model(agent, model_directory, new_model_fingerprint)
-    else:
-        logger.debug(f"No new model found at URL {model_server.url}")
+    model_directory = tempfile.mkdtemp()
+    loaded = False
+
+    try:
```

Comment:
what do we need the `try: ` block for?

Reply:
If you expect an unzip error - shouldn't this be handled as part of `_pull_model_and_fingerprint` ?

Reply:
I think handling the creation/deletion should be outside `_pull_model_and_fingerprint`, because after the model is downloaded there is still room for errors to happen. So it is easier to move this management to `_update_model_from_server` and handle it there.

I used try/finally because we only want to delete the directory if an error happened. I would've preferred to use `with tempfile.TemporaryDirectory() as ...`, but that would delete it no matter what.

Reply:
Makes sense. But using this logic `_load_and_set_updated_model` should not have a `try ... except` block then, should it?

Reply:
I think we could let the exception bubble up from `_load_and_set_updated_model`. However then we would need to catch it in `_update_model_from_server` because it at the moment it is being converted into a log message and then ignored.

Reply:
yep, that's what I mean. I think it would be a bit cleaner to handle all exception stuff in one place. Anyway, up to you as we're quite close to the release and it doesn't really matter.